src/containers/LoadFactory.tsx

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright (c) 2018-2020 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+
+import { AlertVariant } from '@patternfly/react-core';
+import React from 'react';
+import { connect, ConnectedProps } from 'react-redux';
+import { History } from 'history';
+import { timeout } from '../services/timeout';
+import { AppState } from '../store';
+import * as FactoryResolverStore from '../store/FactoryResolver';
+import * as WorkspaceStore from '../store/Workspaces';
+import LoadFactoryPage from '../pages/loadFactory';
+import { selectAllWorkspaces, selectWorkspaceById } from '../store/Workspaces/selectors';
+import { WorkspaceStatus } from '../services/workspaceStatus';
+
+const WS_ATTRIBUTES_TO_SAVE: string[] = ['workspaceDeploymentLabels', 'workspaceDeploymentAnnotations'];
+
+type Props =
+  MappedProps
+  & { history: History };
+
+type State = {
+  search?: string;
+  location?: string;
+  devfileLocationInfo?: string;
+  currentStep: number;
+  hasError: boolean;
+};
+
+export class LoadFactory extends React.PureComponent<Props, State> {
+  private loadFactoryPageCallbacks: { showAlert?: (variant: AlertVariant.success | AlertVariant.danger, title: string) => void } = {};
```

Comment:
```suggestion
  private loadFactoryPageCallbacks: { showAlert?: (variant: AlertVariant, title: string) => void } = {};
```

Reply:
Agreed to extend types for the future. I have fixed it.